credentials/credentials.go

```diff
@@ -71,9 +71,15 @@ type Credentials interface {
 // TransportAuthenticator defines the common interface all supported transport
 // authentication protocols (e.g., TLS, SSL) must implement.
 type TransportAuthenticator interface {
-	// Dial connects to the given network address and does the authentication
-	// handshake specified by the corresponding authentication protocol.
-	Dial(addr string) (net.Conn, error)
+	// Dial connects to the given network address using net.Dial and then
+	// does the authentication handshake specified by the corresponding
+	// authentication protocol.
+	Dial(network, addr string) (net.Conn, error)
+	// DialWithDialer connects to the given network address using
+	// dialer.Dialand does the authentication handshake specified by the
+	// corresponding authentication protocol. Any timeout or deadline
+	// given in the dialer apply to connection and handshake as a whole.
+	DialWithDialer(dialer *net.Dialer, network, addr string) (net.Conn, error)
```

Comment:
This is essential part of dial timeout support. It is used to pass timeout when TLS does dialing and handshake. This just mimics the API in the standard tls package (http://golang.org/pkg/crypto/tls/#DialWithDialer).